src/CSharp/CSharp/SyntaxRewriters/WhitespaceRemover.cs

```diff
@@ -42,11 +42,16 @@ public override SyntaxTrivia VisitTrivia(SyntaxTrivia trivia)
         if (trivia.IsKind(SyntaxKind.EndOfLineTrivia))
         {
             // We can only safely remove EndOfLineTrivia if it is not proceeded by a SingleLineComment
-            int triviaIndex = trivia.Token.TrailingTrivia.IndexOf(trivia);
+            SyntaxTriviaList triviaList = (trivia.Token.TrailingTrivia.IndexOf(trivia) == -1)
+                ? trivia.Token.LeadingTrivia
+                : trivia.Token.TrailingTrivia;
+
+            int triviaIndex = triviaList.IndexOf(trivia);
+
```

Comment:
It's not necessary to look twice every time. Only if the trivia is not found in the first list.